share/functions/__fish_print_interfaces.fish

```diff
@@ -3,6 +3,11 @@ function __fish_print_interfaces --description "Print a list of known network in
         set -l interfaces /sys/class/net/*
         string replace /sys/class/net/ '' $interfaces
     else # OSX/BSD
-        command ifconfig -l | string split ' '
+        set -l os (uname)
+        if string match -q "OpenBSD" -- $os
```

Comment:
Does the openbsd output here differ from the macOS/FreeBSD stuff?

What does it look like? Can we make one path that works for both?

Reply:
It does, ya.. Might be possible to make one that works for both.. I am not well versed in 'fish' thought.. Here is the output of a typical system:

```
lo0: flags=8049<UP,LOOPBACK,RUNNING,MULTICAST> mtu 32768
        index 4 priority 0 llprio 3
        groups: lo
        inet6 ::1 prefixlen 128
        inet6 fe80::1%lo0 prefixlen 64 scopeid 0x4
        inet 127.0.0.1 netmask 0xff000000
em0: flags=8802<BROADCAST,SIMPLEX,MULTICAST> mtu 1500
        lladdr ff:ff:ff:ff:ff:ff
        index 1 priority 0 llprio 3
        media: Ethernet autoselect (none)
        status: no carrier
iwm0: flags=8843<UP,BROADCAST,RUNNING,SIMPLEX,MULTICAST> mtu 1500
        lladdr ff:ff:ff:ff:ff:ff
        index 2 priority 4 llprio 3
        groups: wlan egress
        media: IEEE802.11 autoselect (HT-MCS13 mode 11n)
        status: active
        ieee80211: nwid WiFi chan 1 bssid ff:ff:ff:ff:ff:ff 100% wpakey <not displayed> wpaprotos wpa2 wpaakms psk wpaciphers ccmp wpagroupcipher ccmp
        inet 255.255.255.255 netmask 0xffffff00 broadcast 255.255.255.255
```

Reply:
I should also mention there is no `-l` on OpenBSD.

Reply:
Look at the example output [in the FreeBSD quickstart guide](https://www.freebsd.org/doc/en/articles/linux-users/network.html):

```
% ifconfig
em0: flags=8843<UP,BROADCAST,RUNNING,SIMPLEX,MULTICAST> mtu 1500
        options=b<RXCSUM,TXCSUM,VLAN_MTU>
        inet 10.10.10.100 netmask 0xffffff00 broadcast 10.10.10.255
        ether 00:50:56:a7:70:b2
        media: Ethernet autoselect (1000baseTX <full-duplex>)
        status: active
em1: flags=8843<UP,BROADCAST,RUNNING,SIMPLEX,MULTICAST> mtu 1500
        options=b<RXCSUM,TXCSUM,VLAN_MTU>
        inet 192.168.10.222 netmask 0xffffff00 broadcast 192.168.10.255
        ether 00:50:56:a7:03:2b
        media: Ethernet autoselect (1000baseTX <full-duplex>)
        status: active
```

That looks basically like your OpenBSD output. So we could omit the "-l" on both, and just split everywhere.

Note that [the FreeBSD manpage](http://nixdoc.net/man-pages/FreeBSD/ifconfig.8.html) says that "-a" is implied if no other arguments are given. Is that true on OpenBSD as well?